File: 20260704/exos/04_fonction.py
# ============================================================
# GTA — gain de mission
# ============================================================
# Franklin reçoit une récompense de base plus un bonus.
# La fonction calcule le total gagné.
#
# Lance : python3 04_fonction.py
# ============================================================

#   base   : l'argent fixe de la mission
#   bonus  : l'argent bonus (tip, objets volés...)
#
# gain(500, 150)  →  650
# gain(1000, 0)   →  1000
#
# Résultat attendu : gain(500, 150) == 650  et  gain(1000, 0) == 1000
#
# Indice : def + return + addition

def gain(base, bonus):
    return base + bonus


# --- Vérification (ne pas modifier) ---
assert gain(500, 150) == 650, "gain(500, 150) doit retourner 650"
assert gain(1000, 0) == 1000, "gain(1000, 0) doit retourner 1000"
print("✅ Correct !")

File: 20260723/exos/04_fonction.lua
-- ============================================================
-- GTA — distance de mission
-- ============================================================
-- Franklin calcule le temps de trajet : distance / vitesse.
-- La fonction retourne le temps (sans arrondi).
--
-- Lance : lua 04_fonction.lua
-- ============================================================

--   distance  : kilomètres
--   vitesse   : km/h
--
-- temps(120, 40)  →  120 / 40 = 3  →  retourne 3
-- temps(90, 30)   →  90 / 30 = 3   →  retourne 3
--
-- Résultat attendu : temps(120, 40) == 3  et  temps(90, 30) == 3
--
-- Indice : function ... end  +  return distance / vitesse

function temps(distance, vitesse)
    return distance / vitesse
end


-- --- Vérification (ne pas modifier) ---
assert(temps(120, 40) == 3, "temps(120, 40) doit retourner 3")
assert(temps(90, 30) == 3, "temps(90, 30) doit retourner 3")
print("✅ Correct !")
